src/plugins/lanplus/lanplus.c

```diff
@@ -3381,6 +3381,56 @@ ipmi_set_session_privlvl_cmd(struct ipmi_intf * intf)
 	return 0;
 }
 
+static uint8_t
+ipmi_find_best_cipher_suite(struct ipmi_intf *intf)
+{
+	/* 256 cipher suites may seem ridiculous, but get channel cipher suites
+	 * can return a maximum of 512 entries:
+	 *   16 * 0x40 bytes / 2 bytes per entry
+	 * though with 5 byte entries, it is 204, but likely will
+	 * only require fewer than 20 (standard defined cipher suites) */
+	struct cipher_suite_info suites[256];
+	size_t nr_suites = sizeof(suites)/sizeof(suites[0]);
+	/* cipher suite best order is chosen with this criteria:
+	 * hmac-md5 and md5 are BAD; xc4 is bad; aes128 is required
```

Comment:
Yes, it would be possible to just check for 17. But I wanted to make it so an arbitrary priority list could be substituted if other people had different opinions about which cipher suites were important to them.